tests/Doctrine/Tests/DBAL/Functional/StatementTest.php

```diff
@@ -0,0 +1,242 @@
+        );
+        $this->_conn->insert('stmt_longer_results', $row1);
+
+        $stmt = $this->_conn->prepare('SELECT param, val FROM stmt_longer_results ORDER BY param');
+        $stmt->execute();
+        $this->assertArraySubset(array(
+            array('param1', 'X'),
+        ), $stmt->fetchAll(\PDO::FETCH_NUM));
+
+        $row2 = array(
+            'param' => 'param2',
+            'val' => 'A bit longer value',
+        );
+        $this->_conn->insert('stmt_longer_results', $row2);
+
+        $stmt->execute();
+        $this->assertArraySubset(array(
+            array('param1', 'X'),
+            array('param2', 'A bit longer value'),
+        ), $stmt->fetchAll(\PDO::FETCH_NUM));
+    }
+
+    public function testFetchLongBlob()
+    {
+        // make sure memory limit is large enough to not cause false positives,
+        // but is still not enough to store a LONGBLOB of the max possible size
+        $this->iniSet('memory_limit', '4G');
+
+        $sm = $this->_conn->getSchemaManager();
+        $table = new Table('stmt_long_blob');
+        $table->addColumn('contents', 'blob', array(
+            'length' => 0xFFFFFFFF,
+        ));
+        $sm->createTable($table);
+
+        $contents = base64_decode(<<<EOF
+H4sICJRACVgCA2RvY3RyaW5lLmljbwDtVNtLFHEU/ia1i9fVzVWxvJSrZmoXS6pd0zK7QhdNc03z
+lrpppq1pWqJCFERZkUFEDybYBQqJhB6iUOqhh+whgl4qkF6MfGh+s87O7GVmO6OlBfUfdIZvznxn
+fpzznW9gAI4unQ50XwirH2AAkEygEuIwU58ODnPBzXGv14sEq4BrwzKKL4sY++SGTz6PodcutN5x
+IPvsFCa+K9CXMfS/cOL5OxesN0Wceygho0WAXVLwcUJBdDVDaqOAij4Rrz640XlXQmAxQ16PHU63
+iqdvXbg4JOHLpILBUSdM7XZEVDDcfuZEbI2ASaYguUGAroSh97GMngcSeFFFerMdI+/dyGy1o+GW
+Ax5FxfAbFwoviajuc+DCIwn+RTwGRmRIThXxdQJyu+z4/NUDYz2DKCsILuERWsoQfoQhqpLhyhMZ
+XfcknBmU0NLvQArpTm0SsI5mqKqKuFoGc8cUcjrtqLohom1AgtujQnapmJJU+BbwCLIwhJXyiKlh
+MB4TkFgvIK3JjrRmAefJm+77Eiqvi+SvCq/qJahQyWuVuEpcIa7QLh7Kbsourb9b66/pZdAd1voz
+fCNfwsp46OnZQPojSX9UFcNy+mYJNDeJPHtJfqeR/nSaPTzmwlXar5dQ1adpd+B//I9/hi0xuCPQ
+Nkvb5um37Wtc+auQXZsVxEVYD5hnCilxTaYYjsuxLlsxXUitzd2hs3GWHLM5UOM7Fy8t3xiat4fb
+sneNxmNb/POO1pRXc7vnF2nc13Rq0cFWiyXkuHmzxuOtzUYfC7fEmK/3mx4QZd5u4E7XJWz6+dey
+Za4tXHUiPyB8Vm781oaT+3fN6Y/eUFDfPkcNWetNxb+tlxEZsPqPdZMOzS4rxwJ8CDC+ABj1+Tu0
+d+N0hqezcjblboJ3Bj8ARJilHX4FAAA=
+EOF
+    );
+
+        $stmt = $this->_conn->prepare('INSERT INTO stmt_long_blob VALUES(?)');
+        $stmt->bindParam(1, $contents, \PDO::PARAM_LOB);
+        $stmt->execute();
+
+        $stmt = $this->_conn->prepare('SELECT contents FROM stmt_long_blob');
+        $stmt->execute();
+
+        $value = $stmt->fetchColumn();
+
+        // this is a temporary workaround to make the test pass on pdo_sqlsrv as the Statement interface
+        // currently doesn't allow binding variables to the result columns which is required to specify blob encoding
+        // @see https://msdn.microsoft.com/en-us/library/ff754357(v=sql.105).aspx#Anchor_3
+        if ($this->_conn->getDriver()->getName() === 'pdo_sqlsrv') {
+            $value = hex2bin($value);
```

Comment:
@morozov I know you can use `bindColumn()` but it seems wrong that this is required because there simply is no reason why it shouldn't work without that. I somewhat studied the driver source code today and it seems the driver is introspecting the column type on fetching anyways and evaluating the appropriate types but it simply does not seem to work. Also applying the situation to `sqlsrv` it works out of the box, too. There is no need for binding anything there. Considering that both drivers share a large part of the codebase, it should be possible for `pdo_sqlsrv` to achieve the desired result.

If we markthe test as skipped, it will get lost. I suppose we will have a solution for this in the future (either by driver bugfix or workaround) and I would like to be able to track some kind of "todo list" to check through once in a while. We did similar things already. I agree it is not the best solution but PHPUnit doesn't offer anything else for this use case. What do you think?